src/js/common/schemaform/ArrayField.jsx

```diff
@@ -52,36 +39,19 @@ export default class ArrayField extends React.Component {
     this.scrollToRow = this.scrollToRow.bind(this);
   }
 
-  /*
-   * This is a performance optimization to avoid extra renders. Because we mirror
-   * formData in local state, each form data change will trigger two renders: one when
-   * local state is updated and another when that change is reflected in formData. This check
-   * skips the second render if no other props or state has changed
-   */
   shouldComponentUpdate(nextProps, nextState) {
-    const propsWithoutDataUnchanged = deepEquals(_.omit('formData', this.props), _.omit('formData', nextProps));
-    const stateUnchanged = deepEquals(this.state, nextState);
-
-    if (propsWithoutDataUnchanged && stateUnchanged && deepEquals(nextProps.formData, nextState.items)) {
-      return false;
-    }
-
-    return true;
+    return !deepEquals(this.props, nextProps) || nextState !== this.state;
```

Comment:
That's true, but this is to avoid updates to this component when a parent component triggers an update but the props for this component aren't actually different.